pandas/core/internals/array_manager.py

```diff
@@ -659,24 +659,50 @@ def idelete(self, indexer):
 
     def iset(self, loc: Union[int, slice, np.ndarray], value):
         """
-        Set new item in-place. Does not consolidate. Adds new Block if not
-        contained in the current set of items
+        Set new column in-place (replaces (an) existing column(s)).
```

Comment:
I think this is specifically not-inplace.

This will behave differently from the BlockManager implementation.  We should test and/or explicitly comment on that.

Reply:
I mean that it changes the ArrayManager in place (it doesn't return a new ArrayManager). I think that's the same as BlockManager?

For the rest, it *replaces* the array for a column, so yet it explicitly *not-inplace* from the array point of view, as it just gets swapped.

Will try to improve the comment.